2020_advent_of_code/day3/day3.cpp

```diff
@@ -0,0 +1,70 @@
+#include<iostream>
+#include<fstream>
+#include<vector>
+
+/*
+  This method counts all the trees the toboggan would meet as it traverses its slope from the top left to the bottom.
+  
+  right - how many places to the right can the sled move
+  down - how many places downward can the sled move
+  hill = the representation of the hill to traverse
+
+  returns the number of trees intersected when going down the hill with a slope of x units right and y units down.
+*/
+int count_trees(int right, int down, const std::vector<std::string>& hill)
+{
+	int modulus = hill[0].length();
+	int trees = 0;
+	int col = 0;
+	int row = 0;
+
+	while (row < hill.size() - 1)
+	{
+		col += right;
+		col %= modulus; // handles wrap around
+		row += down;
+
+		if (hill[row].at(col) == '#')
+		{
+			trees++;
+		}
+	}
+
+	return trees;
+}
+
+int main(int argc, char** argv)
+{
```

Comment:
May want to write a line processing utility that returns an `std::vector<string>`.